src/pages/Earn/VesterDepositModal.tsx

```diff
@@ -0,0 +1,254 @@
+import { Trans, t } from "@lingui/macro";
+import { ethers } from "ethers";
+import React, { useState } from "react";
+import Vester from "sdk/abis/Vester.json";
+import { getIcons } from "config/icons";
+import { SetPendingTransactions } from "domain/legacy";
+import { bigMath } from "lib/bigmath";
+import { callContract } from "lib/contracts";
+import { getPageTitle } from "lib/legacy";
+import { formatAmount, formatAmountFree, parseValue } from "lib/numbers";
+import { UncheckedJsonRpcSigner } from "lib/rpc/UncheckedJsonRpcSigner";
+import Button from "components/Button/Button";
+import SEO from "components/Common/SEO";
+import Modal from "components/Modal/Modal";
+import BuyInputSection from "components/BuyInputSection/BuyInputSection";
+import StatsTooltipRow from "components/StatsTooltip/StatsTooltipRow";
+import TooltipWithPortal from "components/Tooltip/TooltipWithPortal";
+
+export function VesterDepositModal(props: {
+  isVisible: boolean;
+  setIsVisible: (isVisible: boolean) => void;
+  chainId: number;
+  title: string;
+  maxAmount: bigint | undefined;
+  value: string;
+  setValue: (value: string) => void;
+  balance: bigint | undefined;
+  vestedAmount: bigint | undefined;
+  averageStakedAmount: bigint | undefined;
+  maxVestableAmount: bigint | undefined;
+  signer: UncheckedJsonRpcSigner | undefined;
+  stakeTokenLabel: string;
+  reserveAmount: bigint | undefined;
+  maxReserveAmount: bigint | undefined;
+  vesterAddress: string;
+  setPendingTxns: SetPendingTransactions;
+}) {
+  const {
+    isVisible,
+    setIsVisible,
+    chainId,
+    title,
+    maxAmount,
+    value,
+    setValue,
+    balance,
+    vestedAmount,
+    averageStakedAmount,
+    maxVestableAmount,
+    signer,
+    stakeTokenLabel,
+    reserveAmount,
+    maxReserveAmount,
+    vesterAddress,
+    setPendingTxns,
+  } = props;
+  const [isDepositing, setIsDepositing] = useState(false);
+  const icons = getIcons(chainId);
+
+  let amount = parseValue(value, 18);
+
+  let nextReserveAmount: bigint | undefined = reserveAmount;
+
+  let nextDepositAmount = vestedAmount;
+  if (amount !== undefined && vestedAmount !== undefined) {
+    nextDepositAmount = vestedAmount + amount;
+  }
+
+  let additionalReserveAmount = 0n;
+  if (
+    amount !== undefined &&
+    nextDepositAmount !== undefined &&
+    averageStakedAmount !== undefined &&
+    maxVestableAmount !== undefined &&
+    maxVestableAmount > 0 &&
+    nextReserveAmount !== undefined
+  ) {
+    nextReserveAmount = bigMath.mulDiv(nextDepositAmount, averageStakedAmount, maxVestableAmount);
+    if (reserveAmount !== undefined && nextReserveAmount > reserveAmount) {
+      additionalReserveAmount = nextReserveAmount - reserveAmount;
+    }
+  }
+
+  const getError = () => {
```

Comment:
In this file present some values which are calculated on every render which is redundant, it's better use useMemo

Reply:
done